src/iostream.c

```diff
@@ -480,17 +493,19 @@ Int ReadFromPty2( UInt stream, Char *buf, Int maxlen, UInt block)
     }
   return nread;
 }
-  
-  
-UInt WriteToPty ( UInt stream, Char *buf, Int len )
+
+
+static UInt WriteToPty ( UInt stream, Char *buf, Int len )
 {
     Int         res;
     Int         old;
-/*  struct timeval tv; */
-/*  fd_set      writefds; */
-/*  int retval; */
-    if (len < 0)
+    if (len < 0) {
+      // FIXME: why do allow 'len' to be negative here? To allow
+      // invoking a "raw" version of `write` perhaps? But we don't
+      // seem to use that anywhere. So perhaps get rid of it or
+      // even turn it into an error?!
```

Comment:
I think allowing negative length is obscure and obviously undocumented, so I am all for turning this into an error.